systest/validation/consensus.go

```diff
@@ -0,0 +1,165 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	pb "github.com/spacemeshos/api/release/go/spacemesh/v1"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/systest/cluster"
+)
+
+type ConsensusData struct {
+	Consensus, State []byte
+}
+
+func getConsensusData(ctx context.Context, distance int, node *cluster.NodeClient) *ConsensusData {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	meshapi := pb.NewMeshServiceClient(node)
+	lid, err := meshapi.CurrentLayer(ctx, &pb.CurrentLayerRequest{})
+	if err != nil {
+		return nil
+	}
+	layers, err := meshapi.EpochNumLayers(ctx, &pb.EpochNumLayersRequest{})
+	if err != nil {
+		return nil
+	}
+	target := int(lid.Layernum.Number) - distance
+	if target < 2*int(layers.Numlayers.Value) {
+		// empty strings are always in consensus
+		return &ConsensusData{}
+	}
+	ls, err := meshapi.LayersQuery(ctx, &pb.LayersQueryRequest{
+		StartLayer: &pb.LayerNumber{Number: uint32(target)},
+		EndLayer:   &pb.LayerNumber{Number: uint32(target)},
+	})
+	if err != nil {
+		return nil
+	}
+	if len(ls.Layer) != 1 {
+		return nil
+	}
+	layer := ls.Layer[0]
+	return &ConsensusData{Consensus: layer.Hash, State: layer.RootStateHash}
```

Comment:
i removed hash in https://github.com/spacemeshos/go-spacemesh/pull/4026.
the reason being that layer hash wasn't properly updated when block validity changes, and it's not used anywhere.
do you think you need it, or can i use aggregated hash instead? i think that represent consensus better.

Reply:
![image](https://user-images.githubusercontent.com/1964576/216942067-4450b604-fbc6-4de5-94d1-f8bce3fb78ac.png)

i don't see aggregated hash

Reply:
didn't get your response in time. it's now left as empty.
will do a followup PR. will leave the filed name as Hash but populate it with the aggregated hash.

Reply:
https://github.com/spacemeshos/go-spacemesh/pull/4046